### PR TITLE
runtime-rs: Add network support for cloud-hypervisor

### PR DESCRIPTION
**Pull request by amshinde**
This PR adds support for adding a network device before starting the cloud-hypervisor VM.

Support for adding and removing network devices is not really added to the resource manager, so supporting this for cloud-hypervisor is not scoped in this PR.

This also changes "pending_devices" for clh implementation from an Option of vector to simply a vector. This simplifies the structure a bit as we can simple iterate over the pending devices instead of having to check for a "Some" value as this is not really required.

Fixes: #6333

### CODE REVIEW

File: src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs [hunk 1]
```diff
@@ -161,32 +152,53 @@ impl CloudHypervisorInner {
...
+    pub(crate) async fn get_shared_devices(
+        &mut self,
+    ) -> Result<(Option<Vec<FsConfig>>, Option<Vec<NetConfig>>)> {
+        let mut shared_fs_devices = Vec::<FsConfig>::new();
+        let mut network_devices = Vec::<NetConfig>::new();
+
+        while let Some(dev) = self.pending_devices.pop() {
+            match dev {
+                DeviceType::ShareFs(dev) => {
+                    let settings = ShareFsSettings::new(dev.config, self.vm_path.clone());
```
Comment (fidencio) [inline src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs:164]: @amshinde, with the go runtime we're passing an FD to be used, not doing so will make SELinux integration way harder in the future.

Would you mind to follow what's done there?

File: src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs [hunk 2]
```diff
@@ -161,32 +152,53 @@ impl CloudHypervisorInner {
...
+impl TryFrom<NetworkConfig> for NetConfig {
+    type Error = anyhow::Error;
 
-                        root_devices.push(fs_cfg);
-                    }
-                    _ => continue,
-                };
-            }
+    fn try_from(cfg: NetworkConfig) -> Result<Self, Self::Error> {
+        let mut net_config = NetConfig {
```
Comment (jodh-intel) [inline src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs:186]: - If set, should / can `NetworkConfig.virt_iface_name` be used for `net_config.id`?

Comment (amshinde) [reply src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs:186]: This is strictly not required, as the agent relies on the mac address solely and not on the interface name for updating the interfaces later within the VM. But no harm updating this.

File: src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs [hunk 3]
```diff
@@ -161,32 +152,53 @@ impl CloudHypervisorInner {
...
+            queue_size: cfg.queue_size as u16,
+            ..Default::default()
+        };
 
-            Ok(Some(root_devices))
+        if let Some(mac) = cfg.guest_mac {
+            net_config.mac = MacAddr { bytes: mac.0 };
         } else {
-            Ok(None)
+            return Err(anyhow!("Missing mac address for network device"));
```
[the remark this answers is not in the document]
Comment (amshinde) [reply src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs]: I have added test for checking this.

File: src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs (hunk 1 above)
Comment (amshinde) [reply src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs:164]: Sure, i'll follow up on this.

**Review by cmaf — APPROVED** [review]
thanks @amshinde

**Pull request merged**